fe/fe-core/src/main/java/com/starrocks/connector/jdbc/ClickhouseSchemaResolver.java

```diff
@@ -0,0 +1,387 @@
+                    try (ResultSet tableSet = connection.getMetaData().getTables(catalogSchema, null, null, null)) {
+                        while (tableSet.next()) {
+                            String tableName = tableSet.getString("TABLE_NAME");
+                            if (tableName.equalsIgnoreCase(partitionInfoTable)) {
+                                return this.supportPartitionInformation = true;
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new StarRocksConnectorException(e.getMessage());
+        }
+        return this.supportPartitionInformation = false;
+    }
+
+    private String getTableNameForPartInfo() {
+        return properties.getOrDefault(KEY_FOR_TABLE_NAME_FOR_PARTITION_INFO, DEFAULT_TABLE_NAME_FOR_PARTITION_INFO);
+    }
+
+    private String getTableNameForTableInfo() {
+        return properties.getOrDefault(KEY_FOR_TABLE_NAME_FOR_TABLE_INFO, DEFAULT_TABLE_NAME_FOR_TABLE_INFO);
+    }
+
+    @Override
+    public Type convertColumnType(int dataType, String typeName, int columnSize, int digits) {
+
+        PrimitiveType primitiveType;
+        boolean isUnsigned = typeName.toLowerCase().startsWith("uint");
+
+        switch (typeName) {
+            case "Int8":
+                primitiveType = PrimitiveType.TINYINT;
+                break;
+            case "UInt8":
+            case "Int16":
+                primitiveType = PrimitiveType.SMALLINT;
+                break;
+            case "UInt16":
+            case "Int32":
+                primitiveType = PrimitiveType.INT;
+                break;
+            case "UInt32":
+            case "Int64":
+                primitiveType = PrimitiveType.BIGINT;
+                break;
+            case "UInt64":
+            case "Int128":
+            case "UInt128":
+            case "Int256":
+            case "UInt256":
+                primitiveType = PrimitiveType.LARGEINT;
+                break;
+            case "Float32":
+                primitiveType = PrimitiveType.FLOAT;
+                break;
+            case "Float64":
+                primitiveType = PrimitiveType.DOUBLE;
+                break;
+            case "Bool":
+                primitiveType = PrimitiveType.BOOLEAN;
+                break;
+            case "String":
+                return ScalarType.createVarcharType(65533);
+            case "Date":
+                primitiveType = PrimitiveType.DATE;
+                break;
+            case "DateTime":
+                primitiveType = PrimitiveType.DATETIME;
+                break;
+            default:
+                // Decimal(9,9), first 9 is precision, second 9 is scale
+                if (typeName.startsWith("Decimal")) {
+                    String[] precisionAndScale =
+                            typeName.replace("Decimal", "").replace("(", "")
```

Comment:
do you have to call `replaceAll`?

Reply:
I am trying to extract the precision and scale here, and using `replace` is just a convenient way to get them. Another option would be using regular expression. What's the concern here?